### DISCUSSION TITLE
Comparing Time Series Model Performance | Customization

### DISCUSSION THREAD

**Discussion by ngupta23** [pycaret/pycaret#1810] [FAQs]
# Background

In previous FAQ's **[here](https://github.com/pycaret/pycaret/discussions/1808)** and **[here](https://github.com/pycaret/pycaret/discussions/1809)**, we saw how easy it is to consistently compare the performance of multiple time series models in `pycaret` and to customize the comparison process to meet performance / time budget tradeoffs.

In this FAQ, we will see how we can customize the comparison further. In particular, we will see how we can include/exclude a subset of models from our analysis. This can be helpful for example when

1. You want to include some slow models but not others
2. You want to exclude some models that you do not trust or know don't work well with your datasets
3. Include only a certain category of models in your comparison, example, only classical statistical models

NOTE: A complete notebook example for comparing models can be **[found here](https://gist.github.com/ngupta23/f94b2f3ce22b7a0203446b77f2972ffa)**.

# Including and Excluding Models

Here we include `prophet` in the comparison, but exclude other slower models such as `bats` and `tbats`.

```python
all_baseline_some_models = exp.compare_models(turbo=False, exclude=['bats', 'tbats'])
```
![image](https://user-images.githubusercontent.com/33585645/140611343-0539048d-42ce-4eec-9a21-5d4eea5faeb6.png)

That is how easy it is to customize the models that you want to include/exclude in your comparison. That's it for this FAQ. Be sure to read the suggested next reads to see how you can tune your models once you get the best one back from `compare_models`. Happy Forecasting!

# Suggested Next Reads

- https://github.com/pycaret/pycaret/discussions/1791
- https://github.com/pycaret/pycaret/discussions/1795

**Comment by moezali1** [comment]
PyCaret 4.0 sunset the legacy time-series module (sktime/prophet integration), so this FAQ no longer maps to a supported code path.

If you still need time-series forecasting, use sktime / prophet / statsforecast directly — those projects have surpassed what the PyCaret wrapper offered.

For 4.0 docs, see https://pycaret.org/docs and the cleanup notice at https://github.com/pycaret/pycaret/issues/4177.

Closing as outdated.
